### PR TITLE
Adds flag to preserve joint order inside `JointActionCfg` action term

### LINKED ISSUE #786: [Proposal] Config action order
### Proposal

I propose allowing the option to maintain the order of joint names in the action configuration when calling mdp.JointPositionActionCfg for example.

### Motivation

The default joint action order does not always align with the order required by the user. Additionally, since it is possible to define a custom joint order in the observation, it would be beneficial to also allow this customization for actions.

### Additional context
image
image (3)

### PR DESCRIPTION
**Pull request by xav-nal**
# Description

Add a new feature to allow the option to maintain the order of joint names in the action configuration when calling mdp.JointPositionActionCfg for example.

In action_cfg.py
```python
    preserve_order:  bool = False
    """Whether to preserve the order of the joint names in the action output. Defaults to False."""
``` 

In joints_actions.py
```python
self._joint_ids, self._joint_names = self._asset.find_joints(self.cfg.joint_names, preserve_order=self.cfg.preserve_order)
``` 
Fixes #786

## Type of change

- New feature 

## Screenshots

"action_cfg.py"
Before:
image

After
image

In "joints_actions.py"
Before:
image

After:
image

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's config/extension.toml file
- [x] I have added my name to the CONTRIBUTORS.md or my name already exists ther

### CODE REVIEW

**Review by Mayankm96 — APPROVED** [review]
Looks good. Could you please update the version in the `changelog` and `extension.toml` as well?

**Review by Dhoeller19 — APPROVED** [review]

**Comment by xav-nal** [comment]
I have just updated the two files changelog and extension.toml

File: source/extensions/omni.isaac.lab/docs/CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,12 @@
 Changelog
 ---------
+0.20.9 (2024-08-06)
+~~~~~~~~~~~~~~~~~~~
+
+Added
+^^^^^
+
+* Added flag to preserve joint order inside JointActionCfg action term.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/docs/CHANGELOG.rst]:
```suggestion
---------

0.20.9 (2024-08-06)
~~~~~~~~~~~~~~~~~~~

Added
^^^^^

* Added a flag to preserve joint ordering inside the :class:`omni.isaac.lab.envs.mdp.JointAction` action term.
```

**Review by Mayankm96 — APPROVED** [review]

**Comment by Mayankm96** [comment]
@xav-nal The checks are failing for the pre-commit. Could you please run the command `./isaaclab.sh -f` and also complete the checklist in the MR template?

**Comment by xav-nal** [comment]
When I run the commands listed here (https://isaac-sim.github.io/IsaacLab/source/refs/contributing.html) to contribute to the documentation, everything works: I see my change updated in the documentation. However I do not see any additional files to include in the pull request. I assume this is normal.

Regarding the point "I have added tests that prove my fix is effective or that my feature works," I must admit that I am not sure how to do do that.

**Comment by Dhoeller19** [comment]
Thanks a lot @xav-nal!

**Pull request merged**